chipsec/logger.py

```diff
@@ -381,87 +401,27 @@ def logger():
 # Hex dump functions
 ##################################################################################
 
+def chunk (array, width = 8):
+    return [ array[i: i + width] for i in range(0, len(array), width) ]
+
 def dump_buffer( arr, length = 8 ):
     """Dumps the buffer."""
-    tmp=[]
-    tmp_str=[]
-    i=1
-    for c in arr:
-        tmp+=["{:02x} ".format(ord(c))]
-        if c in string.whitespace or c not in string.printable:
-            ch = " "
-        else:
-            ch = ord(c)
-        tmp_str+=["%c"%ch]
-        if i%length==0:
-            tmp+=["| "]
-            tmp+=tmp_str
-            tmp+=["\n"]
-            tmp_str=[]
-        i+=1
-    if 0 != len(arr)%length:
-        tmp+=[ (length - len(arr)%length) * 3*" " ]
-        tmp+=["| "]
-        tmp+=tmp_str
-        tmp+=["\n"]
-    return "".join(tmp)
+
+    result = ""
+    pieces = chunk(arr, length)
+    for piece in pieces:
+        left = "".join(["{:02x} ".format(c) for c in piece])
+        right = "".join(["{:s} ".format(chr(c)) if chr(c) in string.printable else '. ' for c in piece ])
+        line = "{:{str_width}s} | {:s}\n".format(left, right, str_width = length * 3)
+        result += line
+
+    return result
+
 
 def print_buffer( arr, length = 16 ):
     """Prints the buffer."""
-    tmp=[]
-    tmp_str=[]
-    i=1
-    for c in arr:
-        tmp+=["{:2x} ".format(ord(c))]
-        if (not c in string.printable) or (c in string.whitespace):
-            ch = " "
-        else:
-            ch = ord(c)
-        tmp_str+=["%c"%ch]
-        if i%length==0:
-            tmp+=["| "]
-            tmp+=tmp_str
-            tmp_s = "".join(tmp)
-            logger().log( tmp_s )
-            tmp_str=[]
-            tmp=[]
-        i+=1
-
-    if 0 != len(arr)%length:
-        tmp+=[ (length - len(arr)%length) * 3*" " ]
-        tmp+=["| "]
-        tmp+=tmp_str
-        tmp_s = "".join(tmp)
-        logger().log( tmp_s )
-
-def print_buffer_bytes( arr, length = 16 ):
-    """Prints the buffer."""
-    tmp=[]
-    tmp_str=[]
-    i=1
-    for c in arr:
-        tmp+=["{:2x} ".format(c)]
-        if (not chr(c) in string.printable) or (chr(c) in string.whitespace):
-            ch = " "
-        else:
-            ch = c
-        tmp_str+=["%c"%ch]
-        if i%length==0:
-            tmp+=["| "]
-            tmp+=tmp_str
-            tmp_s = "".join(tmp)
-            logger().log( tmp_s )
-            tmp_str=[]
-            tmp=[]
-        i+=1
-
-    if 0 != len(arr)%length:
-        tmp+=[ (length - len(arr)%length) * 3*" " ]
-        tmp+=["| "]
-        tmp+=tmp_str
-        tmp_s = "".join(tmp)
-        logger().log( tmp_s )
-
+    res = dump_buffer(arr, length)
```

Comment:
The changes are causing python2 to break.  For example bios_kbrd_buffer.  It is also breaking the util commands, for example chipsec_util.py acpi table UEFI

Reply:
bummer.  Will need to find another way to achieve the same being compatible